cob_twist_controller/src/callback_data_mediator.cpp

```diff
@@ -0,0 +1,89 @@
+/*!
+ *****************************************************************
+ * \file
+ *
+ * \note
+ *   Copyright (c) 2014 \n
+ *   Fraunhofer Institute for Manufacturing Engineering
+ *   and Automation (IPA) \n\n
+ *
+ *****************************************************************
+ *
+ * \note
+ *   Project name: care-o-bot
+ * \note
+ *   ROS stack name: cob_control
+ * \note
+ *   ROS package name: cob_twist_controller
+ *
+ * \author
+ *   Author: Marco Bezzon, email: Marco.Bezzon@ipa.fraunhofer.de
+ *
+ * \date Date of creation: June, 2015
+ *
+ * \brief
+ *   Methods implementation of the callback data mediator.
+ *   Collects data from ROS callbacks and provides the data
+ *   for constraint parameters.
+ *
+ ****************************************************************/
+#include <ros/ros.h>
+
+#include "cob_twist_controller/callback_data_mediator.h"
+
+CallbackDataMediator::CallbackDataMediator()
```

Comment:
This class provides data for the constraints classes. 

From one side it receives data on the other side it fills the constraint params classes. That's the reason why it is a called a mediator between two classes. 

Additionally a clean solution because the callback is separated from the already large cob_twist_controller code.